packages/server/modules/gatekeeper/services/readOnly.ts

```diff
@@ -0,0 +1,24 @@
+import { GetWorkspacePlan } from '@/modules/gatekeeper/domain/billing'
+import { Workspace } from '@/modules/workspacesCore/domain/types'
+import { throwUncoveredError } from '@speckle/shared'
+
+export const isWorkspaceReadOnlyFactory =
+  ({ getWorkspacePlan }: { getWorkspacePlan: GetWorkspacePlan }) =>
+  async ({ workspaceId }: { workspaceId: Workspace['id'] }) => {
+    const workspacePlan = await getWorkspacePlan({ workspaceId })
+    // Should never happen
+    if (!workspacePlan) return true
+
+    switch (workspacePlan.status) {
+      case 'cancelationScheduled':
+      case 'valid':
+      case 'trial':
+      case 'paymentFailed':
```

Comment:
As stated [here](https://docs.stripe.com/billing/subscriptions/overview#failed-payments), stripe marks the subscription as `past_due` after one payment incomplete, which means we mark the workspace_plan as `paymentFailed`.
We might not want to block the workspace at this point but wait for stripe to follow its rules and retry.
After a certain amount of retries and failures the subscription will be marked as cancelled by stripe itself